src/NovelRenderingService.cpp

```diff
@@ -46,99 +51,115 @@ bool NovelRenderingService::initializeRenderPipeline(const int displayNumber) {
   _openGLContext = SDL_GL_CreateContext(_window.get());
   SDL_GL_MakeCurrent(_window.get(), _openGLContext);
   if (!gladLoadGL()) {
-    fprintf(stderr, "Failed to initialize glad\n");
+    std::cerr << "ERROR: Failed to initialise glad." << std::endl;
     return -1;
   }
-  _programID = LoadShaders("BasicVertexShader.glsl", "BasicFragmentShader.glsl");
+
+  std::cout << "GL_VERSION : " << glGetString(GL_VERSION) << std::endl;
+  std::cout << "GL_SHADING_LANGUAGE_VERSION: " << glGetString(GL_SHADING_LANGUAGE_VERSION) << std::endl;
+
+
+
+  glEnable(GL_BLEND);
+  glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+
+  _basicFillRectProgramId = loadShaders("BasicVertexShader.glsl", "BasicFragmentShader.glsl");
+  _texturedRectProgramId = loadShaders("TexturedVertexShader.glsl", "TexturedFragmentShader.glsl");
   return true;
 }
 
-GLuint NovelRenderingService::LoadShaders(const char * vertex_file_path,const char * fragment_file_path){
+GLuint NovelRenderingService::loadShaders(std::string vertexFilePath , std::string fragmentFilePath){
 
   // Create the shaders
-  GLuint VertexShaderID = glCreateShader(GL_VERTEX_SHADER);
-  GLuint FragmentShaderID = glCreateShader(GL_FRAGMENT_SHADER);
+  GLuint vertexShaderId = glCreateShader(GL_VERTEX_SHADER);
+  GLuint fragmentShaderId = glCreateShader(GL_FRAGMENT_SHADER);
 
   // Read the Vertex Shader code from the file
-  std::string VertexShaderCode;
-  std::ifstream VertexShaderStream(vertex_file_path, std::ios::in);
+  std::string vertexShaderCode;
+
+  std::ifstream VertexShaderStream(vertexFilePath, std::ios::in);
   if(VertexShaderStream.is_open()){
     std::stringstream sstr;
     sstr << VertexShaderStream.rdbuf();
-    VertexShaderCode = sstr.str();
+    vertexShaderCode = sstr.str();
     VertexShaderStream.close();
   }else{
-    printf("Impossible to open %s. Are you in the right directory ? Don't forget to read the FAQ !\n", vertex_file_path);
+    std::cerr << "ERROR: Target Vertex Shader file cannot be opened! Please ensure the path is correct and that the file is not locked." << std::endl;
     getchar();
     return 0;
   }
 
   // Read the Fragment Shader code from the file
-  std::string FragmentShaderCode;
-  std::ifstream FragmentShaderStream(fragment_file_path, std::ios::in);
-  if(FragmentShaderStream.is_open()){
-    std::stringstream sstr;
-    sstr << FragmentShaderStream.rdbuf();
-    FragmentShaderCode = sstr.str();
-    FragmentShaderStream.close();
+  std::string fragmentShaderCode;
+  std::ifstream fragmentShaderStream(fragmentFilePath, std::ios::in);

```

Comment:
I'm unsure on this one. We plan to have a folder structure in the future for end-user projects so I guess it depends on how we choose to handle that?

Reply:
We should probably handle this now, seeing as this can already cause launch issues depending on how it's built (i.e. Windows CMake fails since it's not automatically copied to the app dir, but Windows VS2019 builds just work). Any ideas on how to handle?

Reply:
The correct way to handle this would be to probably have some file system handling service for the folders I specified which handles logic like that, but that's out of the scope of this PR I think?

Reply:
I've opened #24 to refer to this, seeing as I agree it's out of PR (I take my previous comment back, don't handle it in this PR).